File: tmp/test_stft_audio.m
fname = 'for_elise_20s.wav';
N = 10000;
NFFT = 2048;
WINDOW = 2048;
NOVERLAP = round(NFFT / 3);
HOP = NFFT - NOVERLAP;
ZERO_PHASE_WINDOWING = 1;

[x, Fs] = audioread(fname);
x = x(1:N);

w = vertcat([0], hanning(WINDOW-2), [0]);
w = w ./ sum(w);

spec = xxspecgram(x, NFFT, Fs, w, NOVERLAP, ZERO_PHASE_WINDOWING);
mX = 20 * log10(abs(spec))';
pX = unwrap(angle(spec))';

File: tmp/test_stft_audio.py
# coding: utf-8

import tools.utilFunctions as UF
import numpy as np
import tools.stft as STFT
from scipy.signal import get_window


fname = 'for_elise_20s.wav';
N = 10000;
NFFT = 2048;
WINDOW = 2048;
NOVERLAP = int(round(NFFT * 1.0 / 3));
HOP = NFFT - NOVERLAP;
ZERO_PHASE_WINDOWING = 1;

Fs, x  = UF.wavread(fname)
x = x[:N]

w = get_window('hanning', WINDOW, False)
mX, pX = STFT.stftAnal(x, w, NFFT, HOP,
                       zero_phase_windowing=ZERO_PHASE_WINDOWING)
print 'mX: {}'.format(mX)
print 'pX: {}'.format(pX)
